inductiva/_cli/pager.py

```diff
@@ -0,0 +1,312 @@
+            else:
+                line += direction
+            self.scroll_to(line)
+
+    def scroll_to(self, line):
+        """Scroll the display to the given line.
+
+        The method scrolls the display so that the given line is at the top
+        of the text pad. The line is clamped to the range [0, line_count-1].
+        """
+        with self._lock:
+            line = min(self._line_count - 1, max(0, line))
+            self._current_line = line
+            self._update_footer()
+
+    def clear(self) -> None:
+        """Clear the content of the file-like object.
+
+        Both the text and footer pads are cleared and the display refreshed.
+        """
+        with self._lock:
+            self._current_line = 0
+            self._line_count = 0
+
+            self._update_footer()
+            self._txtpad.erase()
+
+            self.flush()
+
+    def run(self):
+        """Run the navigation loop"""
+        try:
+            while True:
+                self.flush()
+                ch = self._window.getch()
+
+                if ch in (curses.ascii.ESC, 113):
+                    # q or ESC
+                    break
+                elif ch == curses.KEY_HOME:
+                    self._scroll(self.SCROLL_HOME, page=True)
+                elif ch == curses.KEY_DOWN:
+                    self._scroll(self.SCROLL_DOWN)
+                elif ch == curses.KEY_UP:
+                    self._scroll(self.SCROLL_UP)
+                elif ch in (curses.KEY_LEFT, 98):
+                    # left or b
+                    self._scroll(self.SCROLL_UP, page=True)
+                elif ch in (curses.KEY_RIGHT, curses.ascii.SP):
+                    # right or space
+                    self._scroll(self.SCROLL_DOWN, page=True)
+
+        except KeyboardInterrupt:
+            return
+
+    def write(self, text: str) -> int:
+        """Write string to the file-like object.
+
+        Returns the total number of lines in the display.
+        Note: The display is only updated after a call to the `flush` method.
+        """
+        with self._lock:
+            self._txtpad.addstr(text)
+
+            self._line_count += text.count("\n")
+            current_line = self._line_count - self._txtpad_view_height
+            self._current_line = max(current_line, 0)
+            self._update_footer()
+            return self._line_count
+
+    def writelines(self, lines: Iterable[AnyStr]) -> None:
+        """Write a list of lines to the file-like object.
+
+        Line separators are not added, so it is usual for each of the
+        lines provided to have a line separator at the end.
+        Note: The display is only updated after a call to the `flush` method.
+        """
+        self.write("".join(lines))
+
+    def flush(self) -> None:
+        """Flush the content of the file-like object to the display."""
+        with self._lock:
+            self._header.refresh(0, 0, 0, 0, self.HEADER, self._width)
```

Comment:
Will we always update the header or just the first time? Or given the trouble, it os just easier to just update it always?